cpp/perspective/src/cpp/vocab.cpp

```diff
@@ -73,9 +73,9 @@ t_vocab::get_interned(const char* s) {
     t_sidxmap::iterator iter = m_map.find(s);
 
     t_uindex idx, bidx, eidx;
-    t_uindex len = strlen(s) + 1;
 
     if (iter == m_map.end()) {
+        t_uindex len = strlen(s) + 1;
```

Comment:
Even better would be updating this function's signature to accept a `size_t s_len` and use that here.  The only caller has a `std::string`, and can call `std::string::size()` (which is cheap, constant time) instead of this function making a potentially expensive strlen() call